approaching_emergency_vehicle_plugin/src/approaching_emergency_vehicle_plugin_node.cpp

```diff
@@ -508,12 +648,611 @@ namespace approaching_emergency_vehicle_plugin
+      resp->new_plan.maneuvers.push_back(composeLaneFollowingManeuverMessage(downtrack_progress, stop_maneuver_beginning_downtrack,  
+                            speed_progress, target_speed, current_lanelet.id(), time_progress));
+      
+      downtrack_progress = stop_maneuver_beginning_downtrack;
+      time_progress += getManeuverDuration(resp->new_plan.maneuvers.back(), epsilon_);
+    }
+
+    // Get the starting lanelet ID for the stop and wait maneuver
+    lanelet::Id start_lane_id = current_lanelet.id();
+
+    // Identify the ending lanelet for the stop and wait maneuver
+    while(current_lanelet_ending_downtrack < end_of_route_downtrack){
+      // Get the next lanelet in the current lane if it exists
+      if(!wm_->getMapRoutingGraph()->following(current_lanelet, false).empty()){
+        current_lanelet = wm_->getMapRoutingGraph()->following(current_lanelet, false).front();
+      }
+      else{
+        RCLCPP_WARN_STREAM(rclcpp::get_logger(logger_name), "A following lanelet in the current lane could not be found; returning empty plan");
+        resp->new_plan.maneuvers = {};
+        return;
+      }          
+
+      // Check whether the next lanelet is on the route
+      if(!wm_->getRoute()->contains(current_lanelet)){
+        RCLCPP_WARN_STREAM(rclcpp::get_logger(logger_name), "The next lanelet in the current lane is not on the route; returning empty plan");
+        resp->new_plan.maneuvers = {};
+        return;
+      }
+
+      // Break from loop if lanelet's ending downtrack is after the route end downtrack; this is the maneuver's ending lanelet 
+      current_lanelet_ending_downtrack = wm_->routeTrackPos(current_lanelet.centerline2d().back()).downtrack;
+      if(current_lanelet_ending_downtrack >= end_of_route_downtrack){
+        break;
+      }
+    }
+
+    // Add stop and wait maneuver to maneuver plan
+    resp->new_plan.maneuvers.push_back(composeStopAndWaitManeuverMessage(downtrack_progress, end_of_route_downtrack,
+                  speed_progress, start_lane_id, current_lanelet.id(), stopping_deceleration, time_progress));
+  }
+
+  void ApproachingEmergencyVehiclePlugin::generateRemainInLaneManeuverPlan(
+    carma_planning_msgs::srv::PlanManeuvers::Response::SharedPtr resp,
+    lanelet::ConstLanelet current_lanelet, double downtrack_progress, double current_lanelet_ending_downtrack,
+    double speed_progress, double target_speed, rclcpp::Time time_progress, bool is_slowing_down_for_erv)
+  {
+    RCLCPP_DEBUG_STREAM(rclcpp::get_logger(logger_name), "Generating remain-in-lane maneuver plan");
+
+    // Reset planned lane change flag since plugin will not be planning for an upcoming lane change
+    has_planned_upcoming_lc_ = false;
+
+    // Reduce target_speed if an ERV is actively passing the ego vehicle
+    if(is_slowing_down_for_erv){
+      target_speed = std::max((target_speed - config_.speed_reduction_during_passing), config_.minimum_reduced_speed);
```

Comment:
my question is, according to this equation when the host vehicle's reduced speed is determined based on this config parameter: config_.speed_reduction_during_passing and it does not matter what is the ERV's speed. Is that the design?

Reply:
Sorry about that, I misread your initial comment and was thinking about "CMV" instead of "ERV". Yes, the design is that the reduced ego vehicle target speed is independent of the ERV speed, and is dependent on the current speed limit (with the ego vehicle maintaining its current speed if it is already less than the target reduced speed limit). 

The ERV's speed is not considered because an ERV will not be tracked by the plugin when the ERV is travelling slower than the ego vehicle, since a slower speed for a rear-approaching ERV indicates that the ERV will not pass the ego vehicle.